web/src/server_events_dispatch.js

```diff
@@ -608,7 +614,9 @@ export function dispatch_normal_event(event) {
                         const sub = sub_store.get(stream_id);
                         stream_settings_ui.update_subscribers_ui(sub);
                     }
-
+                    if (overlays.is_user_profile_modal_open()) {
+                        user_profile.update_user_profile_streams_list();
+                    }
                     compose_fade.update_faded_users();
```

Comment:
So I think the code after `peer_data.bulk_remove_subscribers` in this helper should be moved to a `stream_events.process_subscriber_update(user_ids, stream_ids)`, which can also be called in the `peer_add` code path.

You can perhaps do that as a prep commit or mini prep PR, and then introduce a version of `overlays.is_user_profile_modal_open()` with an interface that returns the ID of the user whose profile modal is open, such that this commit would just extend `stream_events.process_subscriber_update` to check if the user ID for the modal is in the list  of `user_ids`, and if so, call your rerender function.